### DISCUSSION TITLE
文档翻译流程重构

### DISCUSSION THREAD

**Discussion by YiiGuxing** [YiiGuxing/TranslationPlugin#5560] [Ideas]
重构文档翻译流程，实现异步非阻塞式文档翻译。

**Comment by xuhuanzy** [comment]
感觉要移除旧接口,  全面使用新接口. 
```text
As of IntelliJ Platform version 2023.1, the Documentation Provider API is deprecated and plugin authors should use the [Documentation Target API](https://plugins.jetbrains.com/docs/intellij/documentation.html#documentation-target-api) instead.
```
刚测了下发现新的`PsiDocumentationTargetProvider`能够捕获到`Rider`系的快速文档(但得按快捷键才能触发),而且 `k2`的文档似乎也不走老的接口了, 目测迟早会把老接口砍掉.

[`https://plugins.jetbrains.com/docs/intellij/documentation.html#documentationtarget`](https://plugins.jetbrains.com/docs/intellij/documentation.html#documentationtarget)

**Reply by YiiGuxing** [reply]
是的，只是一直没有时间处理，而且新 API 对拓展来说好像并不是很友好，，，

**Reply by xuhuanzy** [reply]
确实挺不友好的，感觉可以找官方技术人员要点支持，毕竟这个项目算idea系插件的顶级项目了。

**Comment by xuhuanzy** [comment]
我用ja-netfilter成功hook到了Rider的文档生成, 要不基于这个做下去?
![image](https://github.com/user-attachments/assets/23b9f819-a192-48e3-984e-e23fb6e2cdb6)
![image](https://github.com/user-attachments/assets/0f731bef-3530-4a3d-b1bb-13b2ad7bb0bb)

**Reply by YiiGuxing** [reply]
这个感觉不太靠谱，还不如直接用反射呢。

**Reply by xuhuanzy** [reply]
字节码性能强点, 而且反射拿不到他的对象吧

**Reply by xuhuanzy** [reply]
或者抛开`ja-netfilter`直接在翻译插件里修改字节码?

**Reply by YiiGuxing** [reply]
这个就更不靠谱了，，，

**Reply by xuhuanzy** [reply]
那我单独维护一个子库吧, 看官方的样子我感觉以后不修改字节码是没办法了

**Reply by xuhuanzy** [reply]
@YiiGuxing 可以qq聊下吗? 要点技术支持, 不知道为啥自己开的插件好好地, 这个库的一debug就疯狂进入异常断点

**Reply by YiiGuxing** [reply]
可以啊，435968721
